### PR TITLE
Add pre-push for lint and unittest

### LINKED ISSUE #9688: Add an npm run unittest push hook
Unit tests should never fail and we should stop people from pushing if unit tests fail. 
They are relatively fast to run, and not intermittent.

### PR DESCRIPTION
**Pull request by bbondy**
**Note to the person merging: This is a PR against 0.17.x, so please merge to 0.18.x and master.**

Fix #9688

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [x] New files have MPL2 license header

### CODE REVIEW

File: tools/lib/ignoredPaths.js [hunk 1]
```diff
@@ -3,6 +3,8 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 module.exports = [
+  'pre-comit',
```
Comment (luixxiul) [inline tools/lib/ignoredPaths.js]: `pre-commit`?

Comment (NejcZdovc) [inline tools/lib/ignoredPaths.js]: Pre-commit is a hook which will run some X script (in our case linter) when you try to do `git commit`.

Comment (luixxiul) [inline tools/lib/ignoredPaths.js]: I just thought it was a typo

Comment (NejcZdovc) [inline tools/lib/ignoredPaths.js]: oh :smile:

Comment (luixxiul) [inline tools/lib/ignoredPaths.js]: thanks for the info anyway ;-)

File: package.json [hunk 2]
```diff
@@ -222,11 +223,16 @@
   },
   "pre-commit": {
     "run": [
-      "lint",
```
Comment (luixxiul) [inline package.json]: by removing this I think the linter does not work during rebasing and it complains there is an error, which makes hard to find out which commit introduced the error, and the fix requires another commit unless you are willing to detect that commit. Sorry if I am wrong but I would like to propose to get this back.

**Pull request merged**